src/psyclone/psyir/tools/call_tree_utils.py

```diff
@@ -411,25 +427,36 @@ def _resolve_calls_and_unknowns(self, outstanding_nonlocals,
                           f"unknown symbol '{signature}'.")
                     continue
 
-                if mod_info.contains_routine(str(signature)):
-                    # It is a routine, which we need to analyse for the use
-                    # of non-local symbols:
-                    outstanding_nonlocals.append(("routine", module_name,
-                                                  signature, access_info))
-                    continue
-                # Check if it is a constant (the symbol should always be found,
-                # but if a module cannot be parsed then the symbol table won't
-                # have been populated)
-                sym_tab = \
-                    mod_info.get_psyir().symbol_table
-                try:
-                    sym = sym_tab.lookup(signature[0])
-                    if sym.is_constant:
-                        continue
-                except KeyError:
-                    print(f"Unable to check if signature '{signature}' "
-                          f"is constant.")
+                cntr = mod_info.get_psyir()
+                if not cntr:
+                    print(f"[CallTreeUtils._resolve_calls_and_unknowns] "
+                          f"Cannot get PSyIR for module '{module_name}' - "
+                          f"ignoring unknown symbol '{signature}'.")
                     sym = None
+                else:
+                    psyir = cntr.get_routine_psyir(str(signature))
+                    if psyir:
+                        # It is a routine, which we need to analyse for the use
+                        # of non-local symbols:
+                        outstanding_nonlocals.append(("routine", module_name,
+                                                      signature, access_info))
+                        continue
+                    else:
+                        print(f"[CallTreeUtils._resolve_calls_and_unknowns] "
+                              f"Cannot find a routine '{signature}' in module "
+                              f"'{module_name}' - ignored.")
+                    # Check whether it is a constant (the symbol should always
+                    # be found, but if a module cannot be parsed then the
+                    # symbol table won't have been populated)
+                    sym_tab = cntr.symbol_table
+                    try:
+                        sym = sym_tab.lookup(signature[0])
+                        if sym.is_constant:
+                            continue
+                    except KeyError:
+                        print(f"Unable to check if signature '{signature}' "
+                              f"is constant.")
+                        sym = None
```

Comment:
Not your fault, but do we need `sym` actually? Same for line 435

Reply:
Good spot, gone now.